app/src/main/kotlin/cz/covid19cz/erouska/ui/main/MainActivity.kt

```diff
@@ -116,6 +144,31 @@ class MainActivity :
         customTabHelper.chromePackageName?.let {
             CustomTabsClient.bindCustomTabsService(this, it, customTabsConnection)
         }
+
+        if (intent?.action == Intent.ACTION_RUN) {
+            when (intent.data) {
+                ShortcutsActions.URL_PAUSE -> {
+                    L.d("Received action intent to pause the service")
+
+                    startService(CovidService.pause(this))
+
+                    Handler().postDelayed({
```

Comment:
Not that stoked about this. I'd probably remove the autohide.
Have you tried to add new activity (that can be transparent? or in a form a of a dialog with simple UI) to handle the action?

Reply:
I've tried adding new activity. The problem with blinking was the same. But having a transparent activity - nope, I didn't try to do that :smile: I'll give it a try.